beacon-chain/core/electra/transition_no_verify_sig.go

```diff
@@ -82,16 +81,31 @@ func ProcessOperations(
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get execution requests")
 	}
+	for _, d := range requests.Deposits {
+		if d == nil {
+			return nil, errors.New("nil deposit request")
+		}
+	}
```

Comment:
Shouldn't this validation logic be in ProcessDepositRequests?

Reply:
The point of this PR is to separate out the error. `ProcessDepositRequests` should returns error typed `execReqErr` which that caller can handle it appropiately

Reply:
I disagree with this. ProcessOperations should not be doing any input validations.

Reply:
Discussed offline, this is planned to be reworked in a later change.